api/newsletter.ts

```diff
@@ -0,0 +1,49 @@
+import { NowRequest, NowResponse } from '@vercel/node'
+import fetch from 'node-fetch'
+
+const API_KEY = process.env.ECOMAIL_API_KEY || ''
+const API_URL = 'https://api2.ecomailapp.cz/lists/2/subscribe'
+
+export default async (request: NowRequest, response: NowResponse) => {
+  const body = request.body
+
+  if (!body || !body.email) {
+    response.status(404).send('Email is missing')
+    return
+  }
+
+  const data = JSON.stringify({
+    subscriber_data: {
+      email: body.email,
+      tags: ['web-subscribe-form'],
+    },
+  })
+
+  try {
+    const res = await fetch(API_URL, {
+      method: 'POST',
+      body: data,
+      headers: {
+        key: API_KEY,
+        'Content-Type': 'application/json',
+      },
+    })
+
+    // error from ecomail
+    if (!res.ok) {
+      const resData = await res.json()
```

Comment:
je třeba tu mít await pro res.json()?

Reply:
Vrací to Promise, tak bych to tam raději nechala.

Reply:
v tom případě ok, nevěděl jsem že to vrací promisu.